storage/sql/sql.go

```diff
@@ -74,6 +75,11 @@ var (
 				}
 
 				if err := fn(tx); err != nil {
+					if pqErr, ok := err.(*pq.Error); ok && pqErr.Code.Name() == "serialization_failure" {
```

Comment:
💭 So, the `serialization_failure` error _potentially_ comes back from the `fn(tx)`, and not only from the `Commit` below, as previously thought? Also, do we know if it _ever_ comes back from the `tx.Commit()`?

Reply:
Yes, I saw it come from both. I believe it just comes down to timing.